xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/MicrosoftStream.xml

```diff
@@ -0,0 +1,782 @@
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </defaultCategory>
+      <description>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>description</name>
+        <number>3</number>
+        <prettyName>Macro description</prettyName>
+        <rows>5</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </description>
+      <id>
+        <disabled>0</disabled>
+        <name>id</name>
+        <number>1</number>
+        <prettyName>Macro id</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($uri = $escapetool.xml($wikimacro.parameters.get('uri')))
+#set ($width = $escapetool.xml($wikimacro.parameters.get('width')))
+#set ($height = $escapetool.xml($wikimacro.parameters.get('height')))
+#set ($showinfo = $escapetool.xml($wikimacro.parameters.get('showinfo')))
+#set ($start = $escapetool.xml($wikimacro.parameters.get('start')))
+#set ($autoplay = $escapetool.xml($wikimacro.parameters.get('autoplay')))
```

Comment:
We definitely need escaping in order to prevent code (wiki syntax, including script macros) injection, but why XML escaping? Is the ``msStream`` macro expecting XML escaped values for its parameters? That would be strange.

Now, indeed, we're missing https://jira.xwiki.org/browse/XWIKI-12839 so the current options are:
* perform the escaping yourself, such as https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.8/xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-ui/src/main/resources/Panels/Navigation.xml#L181-L182
* use ``$services.rendering.escape()`` but its behavior might change in the future because it isn't really designed for macro parameters, as Anca pointed out in https://jira.xwiki.org/browse/XWIKI-11979?focusedCommentId=88806&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-88806 .

@tmortagne do you have any recommendation on this?

Reply:
It's very unlikely that `$services.rendering.escape()` changes any time soon, as an optimized escaping would simply be much more complex (i.e. full of bugs) and much slower. As for "will is support wiki macro parameters in the future", since there are no indication of the context of what it's escaping it has to escape anything that can have a meaning in the indicated syntax and wiki macro syntax is wiki syntax.

Reply:
Then I think you should use ``$services.rendering.escape()``, @oanalavinia .